linkerd/app/core/src/metrics/tcp_accept_errors.rs

```diff
@@ -18,16 +18,16 @@ metrics::metrics! {
     }
 }
 
-#[derive(Clone, Debug)]
-pub struct Registry {
+pub struct Registry<L = LabelAcceptErrors> {
     scopes: metrics::SharedStore<OrigDstAddr, Scope>,
     metric: linkerd_error_metrics::Metric,
+    _labels: PhantomData<fn(L)>,
 }
 
 type Scope = Mutex<HashMap<AcceptErrors, metrics::Counter>>;
 
-type NewErrorMetrics<N, S> =
-    metrics::NewMetrics<N, OrigDstAddr, Scope, RecordError<LabelAcceptErrors, AcceptErrors, S>>;
+type NewErrorMetrics<N, S, L> =
+    metrics::NewMetrics<N, OrigDstAddr, Scope, RecordError<L, AcceptErrors, S>>;
```

Comment:
Do we really need to be generic here? It seems like it would be much simpler to simply have a dedicated error type and enum variant (like we do for `TlsDetectTimeout`). What are the tradeoffs for doing it this way?

Reply:
The motivation behind this approach was that I wanted to be able to define all the code relating to `LINKERD2_PROXY_INBOUND_IPS` in the inbound crate, without having to add the error type someplace where both inbound and outbound would have to be aware of it. This might have been a silly preference, though...if you'd prefer, we don't have to do it this way.

Reply:
Ah, yeah. That totally makes sense.

Given that the `inbound_...` metric itself is defined in the core crate (along with all of the metric labels for both inbound and outbound metrics), I'm not sure the conceptual purity is worth the overhead at this point. We're probably due for some deeper investment in cleaning up our metrics infrastructure, but for now I'd probably lean towards simplicity & consistency by taking a more static approach here.

Reply:
:woman_shrugging: fair enough

Reply:
Okay, I've rebased this onto `main` (removing the metrics change) and moved the `allow_ips` filter into `app-core`, so we can handle its error type in `app-core`'s `tcp_accept_error_total` metrics code, rather than composing handling for the `InvalidIp` error type in the `inbound` crate. It's definitely much simpler that way, even if I have a slight ideological preference for doing it the other way.